editor/import/audio_stream_import_settings.cpp

```diff
@@ -0,0 +1,652 @@
+			beats_edit->set_value(0);
+			beats_enabled->set_pressed(false);
+			bar_beats_edit->set_value(4);
+			loop->set_pressed(false);
+			loop_offset->set_value(0);
+		}
+
+		_preview_zoom_reset();
+		updating_settings = false;
+		_settings_changed();
+
+		set_title(vformat(TTR("Audio Stream Importer: %s"), p_path.get_file()));
+		popup_centered();
+	}
+}
+
+void AudioStreamImportSettings::_bind_methods() {
+}
+
+void AudioStreamImportSettings::_settings_changed() {
+	if (updating_settings) {
+		return;
+	}
+
+	updating_settings = true;
+	stream->call("set_loop", loop->is_pressed());
+	stream->call("set_loop_offset", loop_offset->get_value());
+	if (bpm_enabled->is_pressed()) {
+		stream->call("set_bpm", bpm_edit->get_value());
+		beats_enabled->show();
+		beats_edit->show();
+		bar_beats_label->show();
+		bar_beats_edit->show();
+		double bpm = bpm_edit->get_value();
+		if (bpm > 0) {
+			float beat_size = 60 / float(bpm);
+			int beat_max = int((stream->get_length() + 0.001) / beat_size);
+			int current_beat = beats_edit->get_value();
+			beats_edit->set_max(beat_max);
+			if (current_beat > beat_max) {
+				beats_edit->set_value(beat_max);
+				stream->call("set_beat_count", beat_max);
+			}
+		}
+		stream->call("set_bar_beats", bar_beats_edit->get_value());
+	} else {
+		stream->call("set_bpm", 0);
+		stream->call("set_bar_beats", 4);
+		beats_enabled->hide();
+		beats_edit->hide();
+		bar_beats_label->hide();
+		bar_beats_edit->hide();
+	}
+	if (bpm_enabled->is_pressed() && beats_enabled->is_pressed()) {
+		stream->call("set_beat_count", beats_edit->get_value());
+	} else {
+		stream->call("set_beat_count", 0);
+	}
+
+	updating_settings = false;
+
+	_preview->update();
+	_indicator->update();
+	color_rect->update();
+}
+
+void AudioStreamImportSettings::_reimport() {
+	params["loop"] = loop->is_pressed();
+	params["loop_offset"] = loop_offset->get_value();
+	params["bpm"] = bpm_enabled->is_pressed() ? double(bpm_edit->get_value()) : double(0);
+	params["beat_count"] = (bpm_enabled->is_pressed() && beats_enabled->is_pressed()) ? int(beats_edit->get_value()) : int(0);
+	params["bar_beats"] = (bpm_enabled->is_pressed()) ? int(bar_beats_edit->get_value()) : int(4);
+
+	EditorFileSystem::get_singleton()->reimport_file_with_custom_parameters(path, importer, params);
+}
+
+AudioStreamImportSettings::AudioStreamImportSettings() {
+	get_ok_button()->set_text(TTR("Reimport"));
+	get_cancel_button()->set_text(TTR("Close"));
+
+	VBoxContainer *main_vbox = memnew(VBoxContainer);
+	add_child(main_vbox);
+
+	HBoxContainer *loop_hb = memnew(HBoxContainer);
+	loop_hb->add_theme_constant_override("separation", 4 * EDSCALE);
+	loop = memnew(CheckBox);
+	loop->set_text(TTR("Enable"));
+	loop->set_tooltip(TTR("Enable looping."));
+	loop->connect("toggled", callable_mp(this, &AudioStreamImportSettings::_settings_changed).unbind(1));
+	loop_hb->add_child(loop);
+	loop_hb->add_spacer();
+	loop_hb->add_child(memnew(Label(TTR("Offset (in sec):"))));
```

Comment:
I suggest removing the unit from the label and adding it as suffix (with translator comment for clarify):
```
loop_hb->add_child(memnew(Label(TTR("Offset:"))));
loop_offset = memnew(SpinBox);
...
// TRANSLATORS: Unit suffix for seconds.
loop_offset->set_suffix("s");
```